File: lc002.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        if(l1 == NULL) return l2;
        if(l2 == NULL) return l1;
        int carry = 0;
        ListNode dummy(0);
        ListNode* res = &dummy;
        while(l1 != NULL || l2 != NULL || carry != 0){
            if(l1 != NULL){
                carry = carry + l1->val;
                l1 = l1->next;
            }
            if(l2 != NULL){
                carry = carry + l2->val;
                l2 = l2->next;
            }
            ListNode* tempNode = new ListNode(carry % 10);
            res->next = tempNode;
            res = res->next;
            carry = carry / 10;
        }
        return dummy.next;
    }
};

File: lc002.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        carry = 0
        dummy = ListNode(0)
        curNode = dummy
        n1 = l1
        n2 = l2
        while(n1 != None or n2 != None or carry):
            curSum = carry
            if(n1 != None):
                curSum += n1.val
                n1 = n1.next
            if(n2 != None):
                curSum += n2.val
                n2 = n2.next
            digit = curSum % 10
            carry = curSum / 10
            newNode = ListNode(digit)
            curNode.next = newNode
            curNode = curNode.next
        return dummy.next
            
